### ISSUE TITLE
go-paramfetch appears to download all params instead of specified size

### ISSUE THREAD

**Issue by cryptonemo** [filecoin-project/go-paramfetch#15]
`filecoin-ffi` uses go-paramfetch in CI in this way and has been for a long time:

```
#!/bin/bash -eo pipefail
DIR=$(pwd)
cd $(mktemp -d)
GOPATH=/tmp GO111MODULE=off go get github.com/filecoin-project/go-paramfetch/paramfetch
GOPATH=/tmp GO111MODULE=off go build -o go-paramfetch github.com/filecoin-project/go-paramfetch/paramfetch
./go-paramfetch 2048 "${DIR}/parameters.json"
```

Now, when it runs, it downloads all of the parameters and takes hours, rather than a few minutes for just the size specified.  This results in repeated CI failures because the parameters downloaded are too large to cache, so environmental CI restores are also not possible.

Example illustrating the issue:

https://app.circleci.com/pipelines/github/filecoin-project/filecoin-ffi/1194/workflows/49734f31-52e1-4bae-a56c-bd4db98ede29/jobs/6724

Another practical issue is that leak checks (that normally take 2-3 hours) timeout on CI since [5 hours](https://circleci.com/docs/2.0/configuration-reference/#jobs) is the max limit (outside of any available configuration).  Since the download now takes 2-3 hours, the leaks checks often fail (timed out).

**Comment by arajasek** [comment]
Fix in #16. Note that this will need a change to the CI invocation you shared above, you now need:

`./go-paramfetch 2048 "${DIR}/parameters.json" "${DIR}/srs-inner-product.json"`

**Comment by cryptonemo** [comment]
Resolved -- much appreciated!
